seqr/management/tests/load_rna_seq_tpm_tests.py

```diff
@@ -62,51 +107,62 @@ def test_command(self, mock_gzip_open, mock_open, mock_logger):
 
-        mock_logger.info.assert_has_calls([
-            mock.call('create 2 RnaSeqTpm for NA19675_D2'),
-            mock.call('create 2 RnaSeqTpm for NA19678_D1'),
-        ])
-        mock_logger.warning.assert_not_called()
+        mock_gzip_file.__next__.return_value = 'sample_id\tgene_id\tTPM\ttissue\n'
+        with self.assertRaises(ValueError) as e:
+            call_command('load_rna_seq_tpm_with_metadata', RNA_FILE_ID)
+        self.assertEqual(str(e.exception), 'Sample NA19675_D2 has no tissue type')
 
-        # Test TPM expression API
-        url = reverse(rna_seq_expression, args=['ENSG00000233750', 'F,M,WB'])
-        self.check_require_login(url)
-        response = self.client.get(url, content_type='application/json')
-        self.assertEqual(response.status_code, 200)
-        self.assertDictEqual(response.json(), {
-            'F': [],
-            'M': [1.04],
-            'WB': [6.04],
-        })
+        mock_gzip_file.__iter__.return_value[0] = 'NA19675_D2\tENSG00000240361\t12.6\tfibroblasts\n'
+        with self.assertRaises(ValueError) as e:
+            call_command('load_rna_seq_tpm_with_metadata', RNA_FILE_ID)
+        self.assertEqual(str(e.exception), 'Mismatched tissue types for sample NA19675_D2: fibroblasts, muscle')
+
+        mock_gzip_file.__iter__.return_value[0] = 'NA19675_D2\tENSG00000240361\t12.6\tmuscle\n'
+        with self.assertRaises(ValueError) as e:
+            call_command('load_rna_seq_tpm_with_metadata', RNA_FILE_ID)
+        self.assertEqual(str(e.exception), 'Unable to find matches for the following samples: NA19677, NA19678_D1')
+
+        mock_open.return_value.__enter__.return_value.__iter__.return_value = ['NA19678_D1\tNA19678']
+        with self.assertRaises(ValueError) as e:
+            call_command('load_rna_seq_tpm_with_metadata', RNA_FILE_ID, '--mapping-file', MAPPING_FILE_ID)
+        self.assertEqual(str(e.exception), 'Unable to find matches for the following samples: NA19677')
+
+        call_command('load_rna_seq_tpm_with_metadata', RNA_FILE_ID, '--mapping-file', MAPPING_FILE_ID,
+                     '--ignore-extra-samples')
+
+        self._assert_expected_command_results(mock_logger)
 
         # Test fails on mismatched tissue
-        mock_open.return_value.__enter__.return_value.__iter__.return_value[2] = 'NA19678_D1\tNA19678\tfibroblasts'
-        call_command('load_rna_seq_tpm', 'new_file.tsv.gz', MAPPING_FILE_ID, '--ignore-extra-samples')
-        mock_logger.warning.assert_called_with('Skipped data loading for the following 1 samples due to mismatched tissue type: NA19678_D1 (fibroblasts to whole_blood)')
+        mock_gzip_file.__iter__.return_value[5] = 'NA19678_D1\tENSG00000233750\t6.04\tfibroblasts\n'
+        call_command('load_rna_seq_tpm_with_metadata', 'new_file.tsv.gz', '--ignore-extra-samples')
+        mock_logger.warning.assert_called_with(EXPECTED_MISMATCHED_TISSUE_WARNING)
```

Comment:
It should need a `mock.logger.reset_mock()` before calling the command.

Reply:
we never had that for the existing test in this file, but I'll add a test that warning is explicitly not called during a successful run

Reply:
No it doesn't. The test has an explicit `mock_logger.warning.assert_not_called()` so we know that on the success run this was not called and therefore this is only testing the failure run